### ISSUE TITLE
Continuous Multipart Streaming

### ISSUE THREAD

**Issue by lganzzzo** [oatpp/oatpp#234]
## Brief

Add the possibility to stream Miltipart data part-by-part.
Example use-case - `multipart/x-mixed-replace` video streaming.

## Progress

Work started in the branch. [better_multipart](https://github.com/oatpp/oatpp/tree/better_multipart)

Example of working `multipart/x-mixed-replace` endpoint:

```cpp
/* Implement custom multipart */
class MPStream : public oatpp::web::mime::multipart::Multipart {
public:
  typedef oatpp::web::mime::multipart::Part Part;
public:

  MPStream()
    : oatpp::web::mime::multipart::Multipart(generateRandomBoundary())
  {}

  std::shared_ptr<Part> readNextPart() override {

    auto part = std::make_shared<Part>();
    part->putHeader(Header::CONTENT_TYPE, "image/jpeg");

    oatpp::String frameData = "<frame-binary-jpeg-data>" // frame data. One jpeg image.

    part->setDataInfo(std::make_shared<oatpp::data::stream::BufferInputStream>(frameData));

    return part;

  }

  void writeNextPart(const std::shared_ptr<Part>& part) override {
    throw std::runtime_error("No writes here!!!");
  }

};

ENDPOINT("GET", "multipart-stream", multipartStream) {
  auto multipart = std::make_shared<MPStream>();
  auto body = std::make_shared<oatpp::web::protocol::http::outgoing::MultipartBody>(
    multipart,
    "multipart/x-mixed-replace",
    true /* flush frames immediately */
  );
  return OutgoingResponse::createShared(Status::CODE_200, body);
}
```

## TODO

- Finalize API. 
- Merge to master.

**Comment by bhorn** [comment]
This looks really nice. Before I did any streaming using websockets and did it "server-driven". Now I can refactor some of them to be "client-driven" and use a simple endpoint! To bad it was just an demo for a fair and I won't be able to work an this particular example :(

**Comment by bhorn** [comment]
Maybe I find some time (later this week) to extract the webcam image-streaming part and refactor it to two examples (streaming with websocket and streaming with mutipart)

**Comment by lganzzzo** [comment]
Hey @bhorn ,
That will be great! 
It will help to polish the API and to test this functionality!

**Comment by lganzzzo** [comment]
Merged to [v1.2.0 branch](https://github.com/oatpp/oatpp/tree/v_1.2.0)
